src/praisonai-code/praisonai_code/cli/commands/run.py

```
@@ -717,6 +717,98 @@ async def _run() -> bool:
+                    f"Reverted session '{session}' conversation to keep "
+                    f"{keep} message(s)."
+                )
+            elif messages:
+                # No index maps to an empty history via revert_to_message
+                # (index 0 keeps one message), so leave history untouched rather
+                # than corrupt it with a ghost message. File revert still stands.
+                output.print_info(
+                    f"Reverted files; session '{session}' conversation left "
+                    "intact (too short to rewind coherently)."
+                )
+        except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)